File: web/ddd-briscola-model/src/lib/input.ts
import {Path} from "./model"
import {Seed, Card as ModelCard} from "./game"

export type MatchKind = string | NumberOfGamesMatchKind | TargetPointsMatchKind

export interface NumberOfGamesMatchKind {
  numberOfMatches:number
}

export interface TargetPointsMatchKind {
  winnerPoints:number
}

export type CompetitionStartDeadline = string | OnPlayerCountDeadline

export interface OnPlayerCountDeadline {
  count:number
  kind:string
}

export interface Player {
  name:string
  password:string  
}

export interface Competition {
  players:Path[]
  kind:MatchKind,
  deadline:CompetitionStartDeadline
}

export interface Card {
  seed:string
  "number":number  
}

export function card(m:ModelCard):Card {
    return {
      number: m.number,
      seed: Seed[m.seed]
    }  
  }

File: ddd-briscola-web/src/main/scala/com/github/gdefacci/briscola/presentation/competition/input.scala
package com.github.gdefacci.briscola.presentation.competition

import org.obl.raz.Path

object Input {

  import com.github.gdefacci.briscola.competition.MatchKind
  import com.github.gdefacci.briscola.competition.CompetitionStartDeadline

  final case class Competition(players: GamePlayers, kind: MatchKind, deadline: CompetitionStartDeadline)

  sealed trait GamePlayers
  final case class Players(players: Set[Path]) extends GamePlayers
  final case class TeamPlayers(players: Set[TeamPlayer], teams:Set[TeamInfo]) extends GamePlayers
  
  final case class TeamPlayer(player:Path, teamName:String)
  final case class TeamInfo(name:String)
  
  final case class Team(name:String, players:Set[Path])
  final case class Teams(teams:Seq[Team]) 
  
}
 